### PR TITLE
Allow devices to set transforms for each camera

### PR DESCRIPTION
**Pull request by asajeffrey**
Allow the device to set different camera transforms for each eye.

### CODE REVIEW

**Comment by asajeffrey** [comment]
Rebased.

**Comment by asajeffrey** [comment]
Rebased. @pcwalton review ping?

**Comment by asajeffrey** [comment]
Rebased.

File: demo/common/src/lib.rs [hunk 1]
```diff
@@ -923,19 +914,48 @@ fn center_of_window(window_size: &WindowSize) -> Point2DF32 {
 
 enum Camera {
     TwoD(Transform2DF32),
-    ThreeD { transform: CameraTransform3D, velocity: Point3DF32 },
+    ThreeD {
+        // The mode will either be 3D or VR
+        mode: Mode,
```
Comment (pcwalton) [inline demo/common/src/lib.rs]: This seems redundant; the mode can be detected by looking at the length of `transforms`. How about making this a method (e.g. `is_stereo()`) on `Camera`?

Comment (asajeffrey) [reply demo/common/src/lib.rs]: Fixed. I kept the `mode()` method, rather than going for `is_stereo()`, just because that fits the way the method is used.

**Comment by asajeffrey** [comment]
Rebased.

**Comment by pcwalton** [comment]
Merged in 3996d9a653f633742010df6aaff6ddca15f2e492.

**Pull request merged**